clientconn.go

```diff
@@ -598,6 +604,14 @@ func init() {
 		panic(fmt.Sprintf("impossible error parsing empty service config: %v", cfg.Err))
 	}
 	emptyServiceConfig = cfg.Config.(*ServiceConfig)
+
+	internal.AddConnectivityStateWatcher = func(cc *ClientConn, w connectivitystate.Watcher) func() {
```

Comment:
This function should use the current state of the `ClientConn` as the initial state for the `Tracker` when creating it, instead of defaulting the initial state to `Idle`. So, this would become something like:

```golang
internal.AddConnectivityStateWatcher = func(cc *ClientConn, w connectivitystate.Watcher) func() {
	if cc.csMgr.connectivityStateTracker == nil {
		cc.csMgr.connectivityStateTracker = connectivitystate.NewTracker(cc.csMgr.getState())
	}
	return cc.csMgr.connectivityStateTracker.AddWatcher(w)
}